offline_signing/transfer.sh

```diff
@@ -177,26 +197,55 @@ case "${CMD}" in
         ;;
 esac
 
-# Walk argument pairs.
-while [[ $# > 0 ]]; do
-    c="$1"
-    v="${2?Must provide a version (e.g. 1234.0.0)}"
-    shift 2
-
-    case "${c}" in
-    -a)
-        $CMD "alpha" "${v}" "amd64-usr"
-        $CMD "alpha" "${v}" "arm64-usr"
-        ;;
-    -b)
-        $CMD "beta" "${v}" "amd64-usr"
-        $CMD "beta" "${v}" "arm64-usr"
-        ;;
-    -s)
-        $CMD "stable" "${v}" "amd64-usr"
-        ;;
-    *)
-        usage
-        ;;
-    esac
-done
+case "${CMD}" in
+    download|upload|ready)
+        # Walk argument pairs.
+        while [[ $# > 0 ]]; do
+            c="$1"
+            v="${2?Must provide a version (e.g. 1234.0.0)}"
+            shift 2
+
+            case "${c}" in
+                -a)
+                    $CMD "alpha" "${v}" "amd64-usr"
+                    $CMD "alpha" "${v}" "arm64-usr"
+                    ;;
+                -b)
+                    $CMD "beta" "${v}" "amd64-usr"
+                    $CMD "beta" "${v}" "arm64-usr"
+                    ;;
+                -s)
+                    $CMD "stable" "${v}" "amd64-usr"
+                    ;;
+                *)
+                    usage
+                    ;;
+            esac
+        done
+    ;;
+    roll)
+        # Walk argument pairs.
+        while [[ $# > 0 ]]; do
+            c="$1"
+            s="${2?Must provide a time in hours}"
+            shift 2
+
+            case "${c}" in
+                -a)
+                    $CMD "alpha" "${s}" "amd64-usr"
+                    $CMD "alpha" "${v}" "arm64-usr"
```

Comment:
`${v}` is not defined.

Since the only difference between `roll` and the other subcommands is the error message for `$2`, I'd suggest making only that one assignment conditional, or making nothing conditional and using a more generic error message.

Reply:
I can make it a more generic error message if `$2` is not assigned. that makes more sense than my current approach.